portal/src/main/scala/cromwell/pipeline/service/UserService.scala

```diff
@@ -0,0 +1,54 @@
+package cromwell.pipeline.service
+
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Directives.complete
+import akka.http.scaladsl.server.Route
+import cromwell.pipeline.datastorage.dao.repository.UserRepository
+import cromwell.pipeline.datastorage.dto.user.UpdateRequest
+import cromwell.pipeline.datastorage.dto.user.{ UpdatePasswordRequest, UpdateRequest }
+import cromwell.pipeline.datastorage.dto.{ User, UserId }
+import cromwell.pipeline.utils.StringUtils
+import play.api.libs.json.JsError
+import play.api.libs.json.JsResult.Exception
+
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.Random
+
+class UserService(userRepository: UserRepository)(implicit executionContext: ExecutionContext) {
+
+  def updateUser(id: String, request: UpdateRequest): Future[Int] = {
+    val updatedUser = User(
+      userId = UserId(id),
+      email = request.email,
+      passwordSalt = "",
+      passwordHash = "",
```

Comment:
Only bad guys do like this.

Consider creating new entity without fields you don't want to change

Reply:
done